### ISSUE TITLE
[Tracking] 0.5 Release

### ISSUE THREAD

**Issue by MichaelBroughton** [tensorflow/quantum#493]
Cirq 0.10.0 is coming sometime in March. Based on past releases TF 2.5.0 should also be coming around that time. We should cut our latest 0.5.0 that depends on both of these. Here are the features we want for the release (assigned people will deliver on them):

1. Implement `tfq.noise` module.  #250 . Implement `sample`, `sampled_expectation` and `expectation`. Owner: @MichaelBroughton
2. Natural gradient op (not necessarily differentiator implementation) xref #380  Owner: @jaeyoo 
3. Inner product op gradients + Fidelity op xref #380  Owner: @jaeyoo  
4. Differentiators get circuits support. Rest of the PRs in this chain here #409 , #429 Owner: @zaqqwerty 
5. Noise tutorial finished #454 (happens after, not needed for release) @MichaelBroughton , @zaqqwerty 
6. Upgrade to Cirq 0.10, TF 2.5.0 Owner: @MichaelBroughton 
7. Release python 3.9 builds if TF will support it. Owner: @MichaelBroughton 

Let's get ready for TFQ 0.5.0!

**Comment by zaqqwerty** [comment]
Will also need to upgrade tutorials, we should pin versions of TF and TFQ

**Comment by MichaelBroughton** [comment]
We are quickly approaching our 0.5.0 release. All we are really waiting on now is TF 2.5.0 and py39 builds in order to release. I'm going to quickly document some of the PRs here:

1. For noise related functionality we had:
https://github.com/quantumlib/qsim/pull/308, https://github.com/tensorflow/quantum/pull/500, https://github.com/tensorflow/quantum/pull/501, https://github.com/tensorflow/quantum/pull/508, https://github.com/tensorflow/quantum/pull/518, https://github.com/tensorflow/quantum/pull/520, https://github.com/tensorflow/quantum/pull/531,https://github.com/tensorflow/quantum/pull/535, https://github.com/tensorflow/quantum/pull/536,https://github.com/tensorflow/quantum/pull/540, https://github.com/tensorflow/quantum/pull/544,https://github.com/tensorflow/quantum/pull/545, https://github.com/tensorflow/quantum/pull/547,https://github.com/tensorflow/quantum/pull/549,https://github.com/tensorflow/quantum/pull/552
2. We decided to cut natural gradient operation from this release and will push it back to a later date.
3. For fidelity and inner product we had:
https://github.com/tensorflow/quantum/pull/554, https://github.com/tensorflow/quantum/pull/474 and https://github.com/tensorflow/quantum/pull/477
4. For the differentiator seperation we had: 
https://github.com/tensorflow/quantum/pull/409, https://github.com/tensorflow/quantum/pull/429, https://github.com/tensorflow/quantum/pull/522, https://github.com/tensorflow/quantum/pull/532, https://github.com/tensorflow/quantum/pull/533 and https://github.com/tensorflow/quantum/pull/537
5. Nois tutorial is still underway and pending review here #454
6. Cirq was bumped here: https://github.com/tensorflow/quantum/pull/498
7. We are still waiting on TF 2.5.0 and py 3.9 coming with them. ( Will post more on this issue once the outstanding points are finished.)
